ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp

```diff
@@ -0,0 +1,232 @@
+  m_simulatedLocationDataSource->setLocationsWithPolyline(Polyline::fromJson(walkingTourPolyineJson), simulationParameters);
+
+  m_mapView->locationDisplay()->setDataSource(m_simulatedLocationDataSource);
+  m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::Recenter);
+  m_mapView->locationDisplay()->setInitialZoomScale(1000);
+  m_mapView->locationDisplay()->start();
+
+  m_simulatedLocationDataSource->start();
+
+  // LocationGeotriggerFeed will be used in instantiating a FenceGeotrigger in createGeotriggerMonitor()
+  m_geotriggerFeed = new LocationGeotriggerFeed(m_simulatedLocationDataSource, this);
+}
+
+// This function is used to create two geotrigger monitors in this sample - one for the sections and one for the points of interest
+// The parameters for this function are the only differences between the two geotrigger monitors
+void Geotriggers::createGeotriggerMonitor(ServiceFeatureTable* serviceFeatureTable, double bufferSize, const QString& geotriggerName)
+{
+  FeatureFenceParameters* featureFenceParameters = new FeatureFenceParameters(serviceFeatureTable, bufferSize, this);
+  // The ArcadeExpression defined in the following FenceGeotrigger returns the value for the "name" field of the feature that triggered the monitor
+  FenceGeotrigger* fenceGeotrigger = new FenceGeotrigger(m_geotriggerFeed, FenceRuleType::EnterOrExit, featureFenceParameters, new ArcadeExpression("$fencefeature.name", this), geotriggerName, this);
+  GeotriggerMonitor* geotriggerMonitor = new GeotriggerMonitor(fenceGeotrigger, this);
+
+  connect(geotriggerMonitor, &GeotriggerMonitor::geotriggerNotification, this, &Geotriggers::handleGeotriggerNotification);
+
+  // Start must be explicitly called. It is called after the signal connection is defined to avoid a race condition in Qt.
+  geotriggerMonitor->start();
+}
+
+void Geotriggers::handleGeotriggerNotification(GeotriggerNotificationInfo* geotriggerNotificationInfo)
+{
+  QScopedPointer<GeotriggerNotificationInfo> scopedNotification(geotriggerNotificationInfo);
+  // GeotriggerNotificationInfo provides access to information about the geotrigger that was triggered
+  QString geotriggerName = geotriggerNotificationInfo->geotriggerMonitor()->geotrigger()->name();
+  QString featureName = geotriggerNotificationInfo->message();
+
+  // FenceGeotriggerNotificationInfo provides access to the feature that triggered the notification
+  FenceGeotriggerNotificationInfo* fenceGeotriggerNotificationInfo = static_cast<FenceGeotriggerNotificationInfo*>(geotriggerNotificationInfo);
+  GeoElement* fenceGeoElement = fenceGeotriggerNotificationInfo->fenceGeoElement();
+
+  ArcGISFeature* fenceFeature = dynamic_cast<ArcGISFeature*>(fenceGeoElement);
+
+  // Set the parent to the Geotriggers QObject so the feature is still available outside the scope of this function
+  fenceFeature->setParent(this);
```

Comment:
This is something I covered with Luke because the features would go out of scope due to the following line.
```cpp
QScopedPointer<GeotriggerNotificationInfo> scopedNotification(geotriggerNotificationInfo);
```
However without that line, `geotriggerNotificationInfo` won't get cleaned up even after `handleGeotriggerNotification()` goes out of scope.